src/request.rs

```diff
@@ -32,6 +33,8 @@ pub struct Request<C: HttpClient + Debug + Unpin + Send + Sync> {
     client: Option<C>,
     /// Holds the state of the request.
     req: Option<http_client::Request>,
+    /// Holds the cookies added to the request.
+    cookies: Option<Vec<Cookie<'static>>>,
```

Comment:
I have opted for a `'static` lifetime here as to not introduce lifetime bounds to the struct. This does however mean that we potentially have to allocate twice in the `set_cookie` method to obtain the required `String`s.